### DISCUSSION TITLE
Make sure message is emitted?

### DISCUSSION THREAD

**Discussion by Oripy** [miguelgrinberg/Flask-SocketIO#1731] [Q&A] [answered]
Hello!

I have an app that sends a message through emit(), then do some calculation that takes some time, then emit() another message.

The issue is that when using uwsgi, both emit() are received by the client after the calculation is performed.

I suppose that is because the server is busy working on the calculation, and is preventing the async function to trigger.
Is there a way to emit() with a blocking function?

Thank you!

**Comment by miguelgrinberg** [comment, marked as answer]
You cannot have code that blocks for a long time, that is incompatible with the async mode that you are using (which I assume is the `gevent_uwsgi` mode).

If your long calculation is done inside of a loop, add `socketio.sleep(0)` inside the loop and that might help.

**Reply by Oripy** [reply]
Thank you, I did add some socketio.sleep(0) and it did allow for smoother communication.
I think I have to find a way to have this long-running process on a separate worker.
